compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -401,23 +401,55 @@ LogicalResult getStaticDims(Operation *op,
   return success();
 }
 
+/// Utility to get the static sizes and strides for `AIEX::NpuDmaMemcpyNdOp`
+/// with implicit addressing.
+LogicalResult getStaticDimsForImplicitAddressing(
+    Operation *op, MemRefType memrefType, SmallVectorImpl<int64_t> &staticSizes,
+    SmallVectorImpl<int64_t> &staticStrides) {
+  // 1. Static sizes.
+  SmallVector<int64_t> shapeArr = llvm::to_vector(memrefType.getShape());
+  std::copy(shapeArr.begin(), shapeArr.end(),
+            staticSizes.end() - shapeArr.size());
+  // 2. Static strides.
+  int64_t strideIndex = staticStrides.size() - 1;
+  // For contiguous access the strides for the last dimension is always 1.
+  int64_t prevStride = 1;
+  for (int64_t i = shapeArr.size() - 1; i >= 0; i--) {
+    staticStrides[strideIndex] = prevStride;
+    prevStride = staticStrides[strideIndex--] * shapeArr[i];
+  }
```

Comment:
I think we should check sizes here as well to ensure `strideIndex` doesn't become negative? And add a testcase for this.